File: php/e10.branch.php
#!/usr/bin/env php
<?php
echo "Input a character: "; $Keypress = fgetc(STDIN);
 
switch (TRUE) {
  case preg_match("/[a-z]/", $Keypress):
    echo "Lowercase letter\n";
    break;
  case preg_match("/[A-Z]/", $Keypress):
    echo "Uppercase letter\n";
    break;
  case preg_match("/[0-9]/", $Keypress):
    echo "Digit\n";
    break;
  default:
    echo "Puncuation, whitespace, or other\n";
}
?>

File: scripts/perl/e10.branch.pl
#!/usr/bin/env perl -w
use v5.10.1;
print "Input a character: "; $keypress=getc(STDIN);

# # multiway branch using given...when construct
given ($keypress) {
  when (/[[:lower:]]/) { print "Lowercase letter\n"; }
  when (/[[:upper:]]/) { print "Uppercase letter\n"; }
  when (/[[:digit:]]/) { print "Digit\n" }
  default { print "Punctuation, whitespace, or other\n"; }
}
